### PR TITLE
feat: add name and namespace validation in the world contract

### LINKED ISSUE #19: feat: add namespace validation in the world contract
Currently only the compiler is actually verifying the invalid character in the namespace.

We should add this at the contract level to avoid any surprise on torii indexing.

### PR DESCRIPTION
**Pull request by remybar**
Fixes #19.

When registering/deploying/upgrading a model/namespace/contract, verify that the name and the namespace follow the Dojo naming rules.

### CODE REVIEW

File: crates/contracts/src/utils/utils.cairo [hunk 1]
```diff
@@ -6,6 +6,35 @@ use core::serde::Serde;
 
 use dojo::model::{Layout, FieldLayout};
 
+#[inline(always)]
```
Comment (glihm) [inline crates/contracts/src/utils/utils.cairo]: We should add some simple cairo tests for those functions. To have a first coverage on the function behavior, and then the one you already did at the contract level.

File: crates/contracts/src/utils/utils.cairo
Comment (glihm) [inline crates/contracts/src/utils/utils.cairo:1]: We might split this `utils.cairo` in different files based on the code, as it's already into a `utils` module. 👍

**Review by glihm — COMMENTED** [review]
Thanks @remybar! Some minor comments, and let's move on. 👍

File: crates/contracts/src/utils/utils.cairo (hunk 1 above)
Comment (remybar) [reply crates/contracts/src/utils/utils.cairo]: I've splitted the `utils.cairo` file and I've added some tests for every util functions 🫡

**Review by glihm — APPROVED** [review]
🚀

**Pull request merged**